src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -143,10 +147,20 @@ const Expander = ({ expanded }: ExpandProps) => {
   return <ExpandIcon $expanded={expanded} />
 }
 
-export function Trade({ trade, outputUSDC, gasUseEstimateUSD, expanded }: TradeProps & TradeTooltip & ExpandProps) {
+export function Trade({
+  trade,
+  outputUSDC,
+  gasUseEstimateUSD,
+  expanded,
+  loading,
+}: TradeProps & TradeTooltip & ExpandProps) {
   return (
     <>
-      <Caption caption={<Price trade={trade} outputUSDC={outputUSDC} />} />
+      <Caption
+        caption={<Price trade={trade} outputUSDC={outputUSDC} />}
```

Comment:
Instead of generalizing to pass `loading` to `Caption` when it is only used by `Trade`, would you consider instead passing `<ThemedText.Body2 opacity={loading ? 0.4 : 1}><Price ... /></ThemedText.Body2>` as the `caption` prop?

Reply:
works for me! i'm glad these can be nested.